[2024-12-28T15:56:58.357+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-12-28T15:56:58.371+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: l30d_dump_pipeline.l30_dump_pipeline manual__2024-12-28T15:51:49.674383+00:00 [queued]>
[2024-12-28T15:56:58.378+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: l30d_dump_pipeline.l30_dump_pipeline manual__2024-12-28T15:51:49.674383+00:00 [queued]>
[2024-12-28T15:56:58.380+0000] {taskinstance.py:2866} INFO - Starting attempt 2 of 2
[2024-12-28T15:56:58.393+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): l30_dump_pipeline> on 2024-12-28 15:51:49.674383+00:00
[2024-12-28T15:56:58.401+0000] {standard_task_runner.py:72} INFO - Started process 9529 to run task
[2024-12-28T15:56:58.404+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'l30d_dump_pipeline', 'l30_dump_pipeline', 'manual__2024-12-28T15:51:49.674383+00:00', '--job-id', '324', '--raw', '--subdir', 'DAGS_FOLDER/L30D_data_dump.py', '--cfg-path', '/tmp/tmpdvgzn49o']
[2024-12-28T15:56:58.406+0000] {standard_task_runner.py:105} INFO - Job 324: Subtask l30_dump_pipeline
[2024-12-28T15:56:58.443+0000] {task_command.py:467} INFO - Running <TaskInstance: l30d_dump_pipeline.l30_dump_pipeline manual__2024-12-28T15:51:49.674383+00:00 [running]> on host 24435936fa97
[2024-12-28T15:56:58.506+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='haziq' AIRFLOW_CTX_DAG_ID='l30d_dump_pipeline' AIRFLOW_CTX_TASK_ID='l30_dump_pipeline' AIRFLOW_CTX_EXECUTION_DATE='2024-12-28T15:51:49.674383+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-28T15:51:49.674383+00:00'
[2024-12-28T15:56:58.508+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-12-28T15:56:58.509+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-12-28T15:56:58.511+0000] {logging_mixin.py:190} INFO - Current task name:l30_dump_pipeline state:running start_date:2024-12-28 15:56:58.373065+00:00
[2024-12-28T15:56:58.511+0000] {logging_mixin.py:190} INFO - Dag name:l30d_dump_pipeline and current dag run status:running
[2024-12-28T15:56:58.512+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-12-28T15:57:02.131+0000] {logging_mixin.py:190} INFO - <pyspark.sql.session.SparkSession object at 0x7f631bd80850>
[2024-12-28T15:57:02.133+0000] {logging_mixin.py:190} INFO - Spark session created successfully
[2024-12-28T15:57:02.138+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:29092 <connecting> [IPv4 ('172.18.0.6', 29092)]>: connecting to kafka:29092 [('172.18.0.6', 29092) IPv4]
[2024-12-28T15:57:02.140+0000] {conn.py:1205} INFO - Probing node bootstrap-0 broker version
[2024-12-28T15:57:02.141+0000] {conn.py:410} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:29092 <connecting> [IPv4 ('172.18.0.6', 29092)]>: Connection complete.
[2024-12-28T15:57:02.246+0000] {conn.py:1267} INFO - Broker version identified as 2.5.0
[2024-12-28T15:57:02.247+0000] {conn.py:1268} INFO - Set configuration api_version=(2, 5, 0) to skip auto check_version requests on startup
[2024-12-28T15:57:02.249+0000] {consumer.py:118} WARNING - group_id is None: disabling auto-commit.
[2024-12-28T15:57:02.250+0000] {subscription_state.py:171} INFO - Updating subscribed topics to: ('L30d',)
[2024-12-28T15:57:02.253+0000] {subscription_state.py:257} INFO - Updated partition assignment: [('L30d', 0)]
[2024-12-28T15:57:02.254+0000] {logging_mixin.py:190} INFO - 0
[2024-12-28T15:57:02.255+0000] {logging_mixin.py:190} INFO - [TopicPartition(topic='L30d', partition=0)]
[2024-12-28T15:57:02.257+0000] {conn.py:380} INFO - <BrokerConnection node_id=1 host=kafka:29092 <connecting> [IPv4 ('172.18.0.6', 29092)]>: connecting to kafka:29092 [('172.18.0.6', 29092) IPv4]
[2024-12-28T15:57:02.258+0000] {conn.py:410} INFO - <BrokerConnection node_id=1 host=kafka:29092 <connecting> [IPv4 ('172.18.0.6', 29092)]>: Connection complete.
[2024-12-28T15:57:02.259+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=kafka:29092 <connected> [IPv4 ('172.18.0.6', 29092)]>: Closing connection. 
[2024-12-28T15:57:02.361+0000] {logging_mixin.py:190} INFO - {TopicPartition(topic='L30d', partition=0): 139}
[2024-12-28T15:57:02.995+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/pipeline/pipeline.py", line 15, in l30d_data_dump_pipeline
    spark_l30d(topic, BROKER)
  File "/opt/airflow/etl/data_dump_v2.py", line 206, in spark_l30d
    .load()
  File "/opt/spark/python/pyspark/sql/readwriter.py", line 314, in load
    return self._df(self._jreader.load())
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
  File "/opt/spark/python/pyspark/errors/exceptions/captured.py", line 185, in deco
    raise converted from None
pyspark.errors.exceptions.captured.IllegalArgumentException: Expected e.g. {"topicA":{"0":23,"1":-1},"topicB":{"0":-2}}, got {TopicPartition(topic='L30d', partition=0): 139}
[2024-12-28T15:57:03.015+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-12-28T15:57:03.017+0000] {logging_mixin.py:190} INFO - Task start:2024-12-28 15:56:58.373065+00:00 end:2024-12-28 15:57:03.015071+00:00 duration:4.642006
[2024-12-28T15:57:03.018+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): l30_dump_pipeline> dag:<DAG: l30d_dump_pipeline> dagrun:<DagRun l30d_dump_pipeline @ 2024-12-28 15:51:49.674383+00:00: manual__2024-12-28T15:51:49.674383+00:00, state:running, queued_at: 2024-12-28 15:51:49.685315+00:00. externally triggered: True>
[2024-12-28T15:57:03.021+0000] {logging_mixin.py:190} INFO - Failure caused by Expected e.g. {"topicA":{"0":23,"1":-1},"topicB":{"0":-2}}, got {TopicPartition(topic='L30d', partition=0): 139}
[2024-12-28T15:57:03.023+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=l30d_dump_pipeline, task_id=l30_dump_pipeline, run_id=manual__2024-12-28T15:51:49.674383+00:00, execution_date=20241228T155149, start_date=20241228T155658, end_date=20241228T155703
[2024-12-28T15:57:03.038+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-12-28T15:57:03.040+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 324 for task l30_dump_pipeline (Expected e.g. {"topicA":{"0":23,"1":-1},"topicB":{"0":-2}}, got {TopicPartition(topic='L30d', partition=0): 139}; 9529)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 417, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/pipeline/pipeline.py", line 15, in l30d_data_dump_pipeline
    spark_l30d(topic, BROKER)
  File "/opt/airflow/etl/data_dump_v2.py", line 206, in spark_l30d
    .load()
  File "/opt/spark/python/pyspark/sql/readwriter.py", line 314, in load
    return self._df(self._jreader.load())
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
  File "/opt/spark/python/pyspark/errors/exceptions/captured.py", line 185, in deco
    raise converted from None
pyspark.errors.exceptions.captured.IllegalArgumentException: Expected e.g. {"topicA":{"0":23,"1":-1},"topicB":{"0":-2}}, got {TopicPartition(topic='L30d', partition=0): 139}
[2024-12-28T15:57:03.075+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-12-28T15:57:03.095+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-12-28T15:57:03.098+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
